docs/basic-configuration.md

```diff
@@ -28,7 +28,7 @@ class PostsTable extends Table
         $this->addBehavior('Search.Search');
 
         // Setup search filter using search manager
-        $this->searchManager()
+        $this->getBehavior('Search')->searchManager()
```

Comment:
But isnt that then executed always? Whereas the current searchManager() is only executed once you actually invoke it?

Reply:
My main concern is the length of initialize() though.
It often already is a lot of lines long for all the relation setup and behavior adding and more.
This adds also the whole search config into it.

So the extra method wasnt too bad after all in terms of readability of the code.
It seems the only alternative really is then to use the collections?

Reply:
I just found out:

Additionally, it breaks all usage of dynamic attachment if one is forced to to abandon this method:

    $this->Users->addBehavior('Search.Search');
    
wouldnt work anymore, so far did by using the method when needed.

Reply:
> But isnt that then executed always?

In the updated example, the `$this->searchManager()` call was also from table's `initialize()` so it too was executed always.

> My main concern is the length of initialize() though.

Yes that's a concern, that is why filter collections were added.

> it breaks all usage of dynamic attachment

Yes it does, the solution is to use filter collections. In general I don't think configuring tables from outside is a good idea.

Anyway, we can undo the deprecation of `searchManager()` on table classes if you prefer.

Reply:
> This was a patch release after all.

It was 1st patch release of a new minor :)

Reply:
Right..^^